packages/server-wallet/src/wallet/store.ts

```diff
@@ -1,14 +1,66 @@
 import Objection from 'objection';
+import {SignedState, calculateChannelId, Objective} from '@statechannels/wallet-core';
 
 import {SignState} from '../protocols/actions';
-import {Channel, SyncState} from '../models/channel';
+import {Channel, SyncState, RequiredColumns} from '../models/channel';
+import {SigningWallet} from '../models/signing-wallet';
+import {logger} from '../logger';
+import {addHash} from '../state-utils';
+import knex from '../db/connection';
 
 export const Store = {
+  transaction: async function(callback: () => Promise<void>): Promise<void> {
+    return knex.transaction(async () => {
+      await callback();
+    });
+  },
   signState: async function(action: SignState, tx: Objection.Transaction): Promise<SyncState> {
     const channel = await Channel.forId(action.channelId, tx);
     const notifications = channel.signAndAdd(action);
     await Channel.query(tx).update(channel);
 
     return notifications;
   },
+
+  addObjective: async function(_objective: Objective): Promise<void> {
+    // TODO: Implement this
+    return Promise.resolve();
+  },
+  addSignedState: async function(signedState: SignedState): Promise<void> {
```

Comment:
Seems like in general this function should be more like:

```ts
const ret = validation(ss);

if (ret is invalid) return <whatever the error was>;

else Channel.upsertSignedState(ss);
```

At the moment the validation and the upsert logic is all in one function.

Reply:
Probably `vet1` should happen _outside` the DB transaction context as well, to minimize time with the lock.

Reply:
Currently `Channel` has a lot of code that looked like it was copied from the previous `store`  that should probably be moved to the current `store` so we can structure the code this way.